src/ast/ast_build_ar_exp.c

```diff
@@ -0,0 +1,328 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "ast_build_ar_exp.h"
+#include "../util/rmalloc.h"
+#include "../arithmetic/funcs.h"
+#include <assert.h>
+
+static const char *_ASTOpToString(AST_Operator op) {
+	switch(op) {
+	case OP_PLUS:
+		return "ADD";
+		break;
+	case OP_MINUS:
+		return "SUB";
+		break;
+	case OP_MULT:
+		return "MUL";
+		break;
+	case OP_DIV:
+		return "DIV";
+		break;
+	case OP_CONTAINS:
+		return "CONTAINS";
+		break;
+	case OP_STARTSWITH:
+		return "STARTS WITH";
+		break;
+	case OP_ENDSWITH:
+		return "ENDS WITH";
+		break;
+	case OP_AND:
+		return "AND";
+		break;
+	case OP_OR:
+		return "OR";
+		break;
+	case OP_XOR:
+		return "XOR";
+		break;
+	case OP_NOT:
+		return "NOT";
+		break;
+	case OP_GT:
+		return "GT";
+		break;
+	case OP_GE:
+		return "GE";
+		break;
+	case OP_LT:
+		return "LT";
+		break;
+	case OP_LE:
+		return "LE";
+		break;
+	case OP_EQUAL:
+		return "EQ";
+		break;
+	case OP_NEQUAL:
+		return "NEQ";
+		break;
+	case OP_MOD:
+		return "MOD";
+		break;
+	case OP_POW:
+		return "POW";
```

Comment:
As a note, all these strings get downcased in `_AR_GetFunc`.

Reply:
yeah, but they would be downcased even if they were lower-cased here.